backend/app.py

```diff
@@ -0,0 +1,113 @@
+"""This app loads the webpage and acts as an API for the RDS."""
+from os import environ as ENV
+
+from dotenv import load_dotenv
+from flask import Flask, request, jsonify
+from utils import get_details_from_post_code
+from utils_db import get_db_connection, get_id, setup_user_location, get_value_from_db, \
+    check_row_exists, get_locations_for_user, update_loc_assignment, delete_user
+
+app = Flask(__name__)
+
+
+@app.route('/login', methods=['POST'])
+def login_user():
+    """This function validates the login of a user."""
+    load_dotenv()
+    conn = get_db_connection(ENV)
+    data = request.json
+    email = data['email'].lower()
+    password = data['password']
+    if not check_row_exists(conn, 'user_details', 'email', email, 'password', password):
+        return jsonify({'error': 'Invalid credentials'}), 400
+
+    _id = get_id('user_details', 'email', email, conn)
+    name = get_value_from_db('user_details', 'name', _id, 'user_id', conn)
+    conn.close()
+    return jsonify({'message': 'Login Successful', 'name': name}), 200
```

Comment:
you could create a global constant for the network error codes like 200. It would cut down the hardcoded numbers